oshi-core/src/main/java/oshi/software/common/AbstractOSThread.java

```diff
@@ -20,12 +20,23 @@ public abstract class AbstractOSThread implements OSThread {
 
     private final Supplier<Double> cumulativeCpuLoad = memoize(this::queryCumulativeCpuLoad, defaultExpiration());
 
+    public void setThreadId(int threadId) {
```

Comment:
Good idea to move the threadId to the superclass with a setter, but these fields are set on instantiation in the constructor and don't change during operation, so this setter is probab better `protected`. 

Also, I see you've implemented inheritance for the 4 Unix OS's and Windows, but don't see Linux or macOS.